### DISCUSSION TITLE
[Kubernetes][Cluster API Provider AWS] Improve observability of CAPA

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#574] [Google Summer of Code 2022]
Cluster API Provider AWS (CAPA) is a subproject of SIG Cluster Lifecycle that extends Cluster API to simplify lifecycle management of Kubernetes clusters on AWS. The focus of this project is to improve the observability of CAPA by integrating it with observability tools such as OpenTelemetry/Jaeger/Prometheus. With the help of metrics/traces emitted by these observability integrations, users will easily analyze CAPA's behaviour and performance. This project will also focus on improving developer experience with OpenTelemetry collectors and by documenting the integration steps.

Mentor(s): Sedef Savas (@sedefsavas) Richard Case (@richcase)

https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#improve-observability-of-capa

**Comment by rvacaru** [comment]
Hi all!
I'm Razvan Vacaru, a data engineer based in Amsterdam. I'm looking into the GSoC 2022 ideas and this is one that I think I'll like and I'll be able to help with. 

In short my cv is:
- Bachelor's degree in CS Engineering (Java, C, Javascript)
- Worked ~6 years mostly as a Java/Spring developer (4 of the 6 years in AWS) 
- Working for ~ 2 years as a Data engineer with Python, terraform, CI/CD pipelines, datadog, ...
- Worked in Golang for the implementation of an AWS datalake (s3 buckets, lambda, sns, rds), for a couple of months
- AWS Developer Associate Certification 
- Limited exposure to K8s, no work experience (1 day hands on training, done some tutorials, watched some videos)

Full CV: https://www.linkedin.com/in/razvan-vacaru-17787b5a/
From time to time I also do small contributions to OS projects like akka or dbt (https://github.com/rvacaru)

My goal is to increase my knowledge of K8s and Golang and AWS as a nice side effect. Do you think I would be a good fit for this project?

On Friday 25th I have a free day at work where I can do whatever I want, if you want I can already start with a very small issue.
